server/command.go

```diff
@@ -10,26 +10,29 @@ import (
 )
 
 const COMMAND_HELP = `* |/welcomebot preview [team-name] [user-name]| - preview the welcome message for the given team name. The current user's username will be used to render the template.
-* |/welcomebot list| - list the teams for which welcome messages were defined`
+* |/welcomebot list| - list the teams for which welcome messages were defined
+* |/welcomebot set_channel_welcome [welcome-message]| - set the welcome message for the given channel. Direct channels are not supported.
+* |/welcomebot get_channel_welcome| - print the welcome message set for the given channel (if any)
+* |/welcomebot delete_channel_welcome| - delete the welcome message for the given channel (if any)
+`
 
 func getCommand() *model.Command {
 	return &model.Command{
 		Trigger:          "welcomebot",
 		DisplayName:      "welcomebot",
 		Description:      "Welcome Bot helps add new team members to channels.",
 		AutoComplete:     true,
-		AutoCompleteDesc: "Available commands: preview, help",
+		AutoCompleteDesc: "Available commands: preview, help, list, set_channel_welcome, get_channel_welcome, delete_channel_welcome",
 		AutoCompleteHint: "[command]",
 	}
 }
 
-func (p *Plugin) postCommandResponse(args *model.CommandArgs, text string) {
-	post := &model.Post{
-		UserId:    p.botUserID,
-		ChannelId: args.ChannelId,
-		Message:   text,
+func responsef(format string, args ...interface{}) *model.CommandResponse {
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         fmt.Sprintf(format, args...),
+		Type:         model.POST_DEFAULT,
 	}
-	_ = p.API.SendEphemeralPost(args.UserId, post)
 }
```

Comment:
Why the is the bot account not longer used to create the post? It seams like an easy UX again to use the bot account instead of the generic system icon:
![Screenshot from 2020-03-22 15-45-47](https://user-images.githubusercontent.com/16541325/77252454-388c6880-6c54-11ea-81ec-f83748d002ff.png)

Reply:
This was introduced once I addressed https://github.com/mattermost/mattermost-plugin-welcomebot/pull/31#discussion_r388489169.

Even if I set the `Username` field of the , the message is still sent as `System`. I wonder if this was the original reason why I keep seeing the pattern of returning an empty `model.CommandResponse{}` objects across plugins.

Let me know if this can be improved somehow.

Reply:
Indeed, the correct pattern is to return `model.CommandResponse{}`  every time in `ExecuteCommand` and use `postCommandResponse` to create the response post.

Reply:
Before I do, @jfrerich could you please confirm that you are OK with the above?? I would like to avoid doing revert of a revert a week later :smiley: